cmd/cluster-provider-gardener/app/app.go

```diff
@@ -56,7 +54,6 @@ func (o *SharedOptions) AddPersistentFlags(cmd *cobra.Command) {
 	// environment
 	cmd.PersistentFlags().StringVar(&o.Environment, "environment", "", "Environment name. Required. This is used to distinguish between different environments that are watching the same Onboarding cluster. Must be globally unique.")
 	// provider name
-	cmd.PersistentFlags().StringVar(&o.ProviderName, "provider-name", "", "Name of the provider resource.")
```

Comment:
Why did you remove the `--provider-name` argument? It is required for the provider to work ...